actions/workflows/archive_upload.yaml

```diff
@@ -0,0 +1,71 @@
+version: "2.0" # mistral version
+name: arteria-packs.archive_upload
+description: Archives a runfolder to PDC with TSM
+
+workflows:
+    main:
+        type: direct
+        input:
+            - runfolder
+            - host
+            - remove_previous_archive
+        output:
+            output_the_whole_workflow_context: <% $ %>
+
+        tasks:
+            ### GENERAL TASKS START ###
+            note_workflow_repo_version:
+              action: core.local
+              input:
+                cmd: git rev-parse HEAD
+                cwd: /opt/stackstorm/packs/arteria-packs/
+              on-success:
+                - get_config
+
+            get_config:
+              action: arteria-packs.get_pack_config
+              publish:
+                archive_upload_port: <% task(get_config).result.result.archive_upload_service_port %>
+              on-success:
+                - create_archive_dir
+
+            ### GENERAL TASKS END ###
+
+            ### TRANSFER FILES START ###
+            create_archive_dir:
+              action: core.http
+              input:
+                url: http://<% $.host %>:<% $.archive_upload_port %>/api/1.0/create_dir/<% $.runfolder %>
+                body: '{"remove": "<% $.remove_previous_archive %>"}'
+                method: "POST"
+              on-success:
+                - compress_tsm_archive
+
+            compress_tsm_archive:
+               action: core.http
+               input:
+                 url: http://<% $.host %>:<% $.archive_upload_port %>/api/1.0/compress_archive/<% $.runfolder %>_archive
+                 method: "POST"
+                 timeout: 3600 # 1 hr timeout
```

Comment:
Perhaps it will take longer than 1 h to compress a large directory here? I'm unsure.

Reply:
Haven't run it on any of the big runfolders, but the previous implementation seems to have taken about 1 minute, so I think it will be fine. It doesn't compress everything.

Reply:
Great!